nca/Utils/QueryOutputHandler.py

```diff
@@ -0,0 +1,358 @@
+# fields that may appear together in one output_explanation and additional info for writing the explanation if required
+@dataclass
+class PoliciesAndRulesExplanations:
+    # used in RedundancyQuery and EmptinessQuery: we may have lists of redundant/empty policies, ingress/egress rules
+    policies_list: list[str] = None
+    policies_to_ingress_rules_dict: dict = None
+    policies_to_egress_rules_dict: dict = None
+
+
+@dataclass
+class PodsListsExplanations:
+    # 2 use cases:
+    # 1. used in ContainmentQuery when a reason that config is not contained in the other,
+    # is pods list that appears only in one
+    # 2. used in AllCapturedQuery - lists of pods that are not captured for ingress and/or egress
+    pods_list: list[str] = None
+    egress_pods_list: list[str] = None
+    add_xgress_suffix: bool = False
+
+
+@dataclass
+class ConnectionsDiffExplanation:
+    # used in TwoNetworkConfigs queries that compare connections of pairs of peers in both configs
+    peers_diff_connections_list: list[tuple] = None
+    additional_description: str = ''
+
+
+@dataclass
+class CombinedExplanation:
+    # used in TwoWayContainment, when both configs do not contain each other -
+    # the output_explanation is a combination of two explanation of different containment queries
+    two_results_combined: list[OutputExplanation] = None
+
+
+class QueryOutputHandler:
+    """
+    A class to handle the output of the query and create it in the correct form ,
+    output explanation is handled by it type
+    """
+    def __init__(self, configs):
+        """
+        :param list[str] configs: list of config(s) name(s)
+        """
+        self.configs_names = configs
+
+    def handle_explanation_by_type(self, explanation):
+        """
+        handles writing the output explanation according to its type
+        :param OutputExplanation explanation: the query's output explanation
+        """
+        if explanation.policies_with_intersect_pods:
+            self.write_policies_with_intersect_pods_explanation(explanation.explanation_description,
+                                                                explanation.policies_with_intersect_pods)
+        elif explanation.policies_and_rules:
+            self.write_policies_and_rules_explanations(explanation.explanation_description,
+                                                       explanation.policies_and_rules)
+        elif explanation.pods_lists:
+            self.write_pods_list_explanation(explanation.explanation_description, explanation.pods_lists)
+        elif explanation.connections_diff:
+            self.write_conns_diff_explanation(explanation.explanation_description, explanation.connections_diff)
+        elif explanation.combined_explanation:
+            self.handle_explanation_by_type(explanation.combined_explanation.two_results_combined[0])
+            self.handle_explanation_by_type(explanation.combined_explanation.two_results_combined[1])
+
```

Comment:
Can we avoid having a separate method for handling each type of explanation? 
Maybe the `OutputExplanation ` object itself can have a method that returns its `txt `or `yaml ` str, and then you can update the relevant member of the OutputHandler with the result.

Reply:
Is it possible to compute the `dict` object from the explanation separately, so then it can be used for creating `yaml `or `json `output?

Reply:
> Is it possible to compute the `dict` object from the explanation separately, so then it can be used for creating `yaml `or `json `output?

yes, my plan when adding "json" support is that JsonOutputHandler , may be derived from the YamlOutputHandler and the to separate the "compute" and "dump" parts so only the dump will be different

Reply:
good, so let's do the separation already on this branch.

Reply:
> Can we avoid having a separate method for handling each type of explanation? Maybe the `OutputExplanation ` object itself can have a method that returns its `txt `or `yaml ` str, and then you can update the relevant member of the OutputHandler with the result.

hi @adisos , for your review, in last commit , each explanation's type dataclass is responsible for computing the yaml and txt outputs (had to write 2 def for each) 
syaml explanation is returned as real "yaml" field (list[dict]) to be added to the other yaml fields and then all together dumped into yaml string